components/camel-resteasy/src/main/java/org/apache/camel/component/resteasy/DefaultResteasyHttpBinding.java

```diff
@@ -0,0 +1,219 @@
+        String body = exchange.getIn().getBody(String.class);
+
+        LOG.debug("Body in producer: " + body);
+
+        String mediaType = exchange.getIn().getHeader(Exchange.CONTENT_TYPE, String.class);
+
+        WebTarget target = client.target(uri);
+
+        LOG.debug("Populate Resteasy request from exchange body: {} using media type {}", body, mediaType);
+
+        Invocation.Builder builder;
+        if (mediaType != null) {
+            builder = target.request(mediaType);
+        } else {
+            builder = target.request();
+        }
+
+
+        for (Map.Entry<String, Object> entry : exchange.getIn().getHeaders().entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            if (headerFilterStrategy != null
+                    && !headerFilterStrategy.applyFilterToCamelHeaders(key, value, exchange)) {
+                builder.header(key, value);
+                LOG.debug("Populate Resteasy request from exchange header: {} value: {}", key, value);
+            }
+        }
+
+        if (parameters.get("username") != null && parameters.get("password") != null) {
+            target.register(new BasicAuthentication(parameters.get("username"), parameters.get("password")));
+        }
+        LOG.debug("Basic authentication was applied");
+        String method = parameters.get("method");
+
+        if (method.equals("GET")) {
+            return builder.get();
+        }
+        if (method.equals("POST")) {
+            return  builder.post(Entity.entity(body, mediaType));
+        }
+        if (method.equals("PUT")) {
+            return  builder.put(Entity.entity(body, mediaType));
+        }
+        if (method.equals("DELETE")) {
+            return  builder.delete();
+        }
+        if (method.equals("OPTIONS")) {
+            return  builder.options();
+        }
+        if (method.equals("TRACE")) {
+            return  builder.trace();
+        }
+        if (method.equals("HEAD")) {
+            return  builder.head();
+        }
+
+        // maybe throw exception because not method was correct
+        throw new IllegalArgumentException("Method '" + method + "' is not supported method");
+    }
+
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Override
+    public void populateProxyResteasyRequestAndExecute(String uri, Exchange exchange, Map<String, String> parameters) {
+        Client client = ClientBuilder.newBuilder().build();
+
+        WebTarget target = client.target(uri);
+
+        if (parameters.get("username") != null && parameters.get("password") != null) {
+            target.register(new BasicAuthentication(parameters.get("username"), parameters.get("password")));
+        }
+
+        if (LOG.isTraceEnabled()) {
+            LOG.trace("Basic authentication was applied");
+        }
+
+        Class realClazz;
+        Object object = null;
+        try {
+            realClazz = Class.forName(parameters.get("proxyClassName"));
```

Comment:
Dont load classes via Class.forName. What is this proxyClassName use-case?

Reply:
i think it is to support proxy framework which is part of resteasy 
https://docs.jboss.org/resteasy/docs/4.5.3.Final/userguide/html/RESTEasy_Client_Framework.html#proxies

Reply:
yes i am aware of the efforts of removing reflections and not sure this feature what to do about this feature, either removing or dropping might be fine as this is a feature in resteasy.

Reply:
So is this mandatory? Do I really need to write a class with annotations to call a remote HTTP REST service?

Reply:
in earlier versions the client framework was a bit limited in resteasy. there are other ways but looks like this is one of the prior methods considering API support coming from resteasy's history. to consider adding / removing such features, engagement with resteasy community might be good but can be said of being not strong ATM

Reply:
https://issues.apache.org/jira/browse/CAMEL-15040